pollux/lib/core/src/main/scala/io/iohk/atala/pollux/core/service/CredentialServiceImpl.scala

```diff
@@ -1048,22 +1075,24 @@ private class CredentialServiceImpl(
         .mapError(RepositoryError.apply)
       size = currentStatusList.size
       lastUsedIndex = currentStatusList.lastUsedIndex
-      // TODO: concurrency issue
-      statusListToBeUsed <-
-        if lastUsedIndex < size then ZIO.succeed(currentStatusList)
-        else
-          credentialStatusListRepository
-            .createNewForTheWallet(jwtIssuer, statusListRegistryUrl)
-            .mapError(RepositoryError.apply)
-
-      _ <- credentialStatusListRepository
-        .allocateSpaceForCredential(
-          issueCredentialRecordId = record.id,
-          credentialStatusListId = statusListToBeUsed.id,
-          statusListIndex = statusListToBeUsed.lastUsedIndex + 1
-        )
-        .mapError(RepositoryError.apply)
+      statusListToBeUsed <- issueCredentialSem.withPermit {
+        for {
+          statusListToBeUsed <-
+            if lastUsedIndex < size then ZIO.succeed(currentStatusList)
+            else
+              credentialStatusListRepository
+                .createNewForTheWallet(jwtIssuer, statusListRegistryUrl)
```

Comment:
I'm not sure if the concurrency issue is resolved, because we do not acquire permit when reading `lastStatusList`, so `lastUsedIndex` might be the same for concurrent operations? IDK, gonna need a second opinion (cc @bvoiturier , @yshyn-iohk )

Maybe a good idea to add test to verify this. You can check [this example](https://github.com/hyperledger-labs/open-enterprise-agent/blob/8f6af25a8eafd27d5017096da64f89188354a2ca/prism-agent/service/wallet-api/src/test/scala/io/iohk/atala/agent/walletapi/service/ManagedDIDServiceSpec.scala#L298).